app/src/main/java/org/thoughtcrime/securesms/onboarding/loading/LoadingViewModel.kt

```diff
@@ -67,10 +67,8 @@ internal class LoadingViewModel @Inject constructor(
 
         viewModelScope.launch(Dispatchers.IO) {
             try {
-                TextSecurePreferences.events
-                    .filter { it == TextSecurePreferences.CONFIGURATION_SYNCED }
-                    .onStart { emit(TextSecurePreferences.CONFIGURATION_SYNCED) }
-                    .filter { prefs.getConfigurationMessageSynced() }
+                prefs.configurationMessageSyncedFlow()
+                    .filter { it }
```

Comment:
Is this just filtering to not null?

Reply:
no, `filter` takes a `predicate: T -> Boolean` and this is a `Flow<Boolean>` so this just filters out falses, cause we're waiting for a `true`, cause we're waiting for the configuration to be synced.

Reply:
Ah, gotcha - thanks.